custom-completions/windows/windows-completions.nu

```diff
@@ -0,0 +1,61 @@
+export module win {
+    # do I have admin rights?
+    export def "am i admin" [] {
+        # https://stackoverflow.com/questions/7985755/how-to-detect-if-cmd-is-running-as-administrator-has-elevated-privileges
+        if (net session | complete).exit_code == 0 {
+            true
+        } else {
+            false
+        }
+    }
+
+    # open in visual studio
+    export def "open in vs-2022" [file: path] {
+        let file = ($file | path expand)
+        run-external `C:\Program Files\Microsoft Visual Studio\2022\Professional\Common7\IDE\devenv.exe` /Edit $file
+    }
+
+    export def "windbg attach-to-process" [
+        --pid(-p): int@"nu-complete processes"  # process-id
+    ] {
+
+        ~/AppData/Local/Microsoft/WindowsApps/WinDbgX.exe -p $pid
+    }
+
+    # sigcheck wrapper, file version and signature viewer
+    export def "which version" [file: path ] {
+        print $"sigcheck -nobanner ($file)"
+        ^sigcheck -nobanner $file | lines | skip 1 | parse --regex '\s*(?<name>.+?):(?<value>.+)'
+    }
+
+    # https://learn.microsoft.com/en-us/sysinternals/downloads/procdump#using-procdump
+    #
+    # Examples:
+    #   Install ProcDump as the (AeDebug) postmortem debugger:
+    #   procdump -ma -i c:\dumps
+    #
+    #   Uninstall ProcDump as the (AeDebug) postmortem debugger:
+    #   procdump -u
+    export extern "procdump" [
+        ...args: any                              # Arguments to be passed to your program
+        # command?: string@"nu-complete rustup"
+    ]
+
+    def "nu-complete proc-names" [] { ps | get name | uniq }
+
+    def "nu-complete ps-priority" [] { [ [value description]; [3 High] [6 'Above Normal'] [2 Normal] [5 'Below Normal'] [1 Low] ] }
+
+    # permanently set priority for process in windows registry
+    export def "ps set-permanent-priority" [
+        proc_name: string@"nu-complete proc-names"
+        --cpu-priority: int@"nu-complete ps-priority" = 2
+        --io-priority: int@"nu-complete ps-priority" = 2
+    ] {
+        # https://answers.microsoft.com/en-us/windows/forum/all/how-to-permanently-set-priority-processes-using/2f9ec439-5333-4625-9577-69d322cfbc5e
+        let perf_options_path = $'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\($proc_name)\PerfOptions'
+        reg add $perf_options_path /v CpuPriorityClass /t REG_DWORD /d $cpu_priority
```

Comment:
I'm not a fan of publishing a script that modifies registry values. It's kind of dangerous and puts the target on our back if someone submits a malicious reg add command that we don't catch.

Reply:
I understand your concern, removed that command, the rest of your comments I will address them the coming week.

Reply:
gone, closing